prov/sockets/src/sock_cntr.c

```diff
@@ -301,6 +304,10 @@ static int sock_cntr_wait(struct fid_cntr *cntr, uint64_t threshold,
 		ret = fi_wait_cond(&_cntr->cond, &_cntr->mut, timeout);
 	}
 
+	if (atomic_get(&_cntr->value) >= threshold &&
+	    atomic_get(&_cntr->last_read_val) < threshold) {
```

Comment:
I guess we can avoid the second check (atomic_get(&_cntr->last_read_val) < threshold)), since application usually calls cntr_read() after a successful fi_poll().